pkg/virt-launcher/virtwrap/manager.go

```diff
@@ -898,14 +899,35 @@ func possibleGuestSize(disk api.Disk) (int64, bool) {
 		return 0, false
 	}
 
-	size := int64((1 - filesystemOverhead) * float64(preferredSize))
+	usableSize, err := getUsableDiskSize(getSourceFile(disk))
+	if err != nil {
+		log.DefaultLogger().Reason(err).Error("Failed to get total usable space, using disk capacity instead")
+		usableSize = *disk.Capacity
+	}
+
+	size := int64((1 - filesystemOverhead) * float64(usableSize))
 	size = kutil.AlignImageSizeTo1MiB(size, log.DefaultLogger())
 	if size == 0 {
 		return 0, false
 	}
 	return size, true
 }
 
+func getUsableDiskSize(path string) (int64, error) {
+	var stat syscall.Statfs_t
+	err := syscall.Statfs(path, &stat)
```

Comment:
So this may get used in the path for block devices as well, not sure, but a failure should show up in e2e if you call statfs on those

Reply:
This function is only used inside `possibleGuestSize`, which is exclusive to fs disks

Reply:
Not sure anymore after seeing so many test failures lol